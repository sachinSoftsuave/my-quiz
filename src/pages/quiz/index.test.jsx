import React from "react";
import { mount } from "enzyme";
import "jsdom-global/register";
import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Quiz from "./index.jsx";
Enzyme.configure({ adapter: new Adapter() });
import toJson from "enzyme-to-json";
import "jest-localstorage-mock";

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));
console.log('hi');
const mockStartQuiz = jest.fn();

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useContext: () => ({
      setQuizStarted: () => mockStartQuiz,
      isQuizStarted: true,
  }),
}));


describe("renders quiz component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Quiz />);
  });

  it('it renders the snapshot of quiz component', () =>{
       expect(toJson(wrapper)).toMatchSnapshot();
  })

});
