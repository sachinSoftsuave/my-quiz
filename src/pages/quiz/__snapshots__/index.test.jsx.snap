// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders quiz component it renders the snapshot of quiz component 1`] = `
<Quiz>
  <div
    className="quizContainer"
  >
    <div
      className="quizModel"
    >
      <h2
        className="quizeQuestion"
      >
        Commonly used data types DO NOT include:
      </h2>
      <Option
        displayAnswer={[Function]}
        index={0}
        option={
          Object {
            "id": "1",
            "value": "strings",
          }
        }
      >
        <div
          className="optionsContainer"
          key="0"
        >
          <p
            className="option"
            onClick={[Function]}
          >
            1. strings
          </p>
        </div>
      </Option>
      <Option
        displayAnswer={[Function]}
        index={1}
        option={
          Object {
            "id": "2",
            "value": "booleans",
          }
        }
      >
        <div
          className="optionsContainer"
          key="1"
        >
          <p
            className="option"
            onClick={[Function]}
          >
            2. booleans
          </p>
        </div>
      </Option>
      <Option
        displayAnswer={[Function]}
        index={2}
        option={
          Object {
            "id": "3",
            "value": "alerts",
          }
        }
      >
        <div
          className="optionsContainer"
          key="2"
        >
          <p
            className="option"
            onClick={[Function]}
          >
            3. alerts
          </p>
        </div>
      </Option>
      <Option
        displayAnswer={[Function]}
        index={3}
        option={
          Object {
            "id": "4",
            "value": "numbers",
          }
        }
      >
        <div
          className="optionsContainer"
          key="3"
        >
          <p
            className="option"
            onClick={[Function]}
          >
            4. numbers
          </p>
        </div>
      </Option>
      <div
        className="resultContainer"
      >
        <p />
      </div>
    </div>
  </div>
</Quiz>
`;
